### PR TITLE
PBI 188: Revamp GetProductByVendor to GetProductVendorByVendor

### PR DESCRIPTION
**Pull request by valeelim**
## Describe your changes
- Convert `GetProductByVendor` to `GetProductVendorByVendor`
- Populate `product_id` field with the respective `product`
- Fixed all related unit tests

## Issue ticket number and link
- [ADUDU-188](https://linear.app/adudu/issue/ADU-188/fix-get-product-vendor-by-vendor-api)

## Postman
image

### CODE REVIEW

File: internal/product/accessor_test.go [hunk 1]
```diff
@@ -447,32 +447,36 @@ func Test_GetProductsByVendor(t *testing.T) {
...
 			WithArgs(vendorID, args.Limit, args.Offset).
 			WillReturnRows(expectedResult)
 
 		totalRows := sqlmock.NewRows([]string{"count"}).AddRow(2)
-		countQuery := `SELECT COUNT(*) FROM product_vendor WHERE vendor_id=$1`
+		countQuery := `SELECT COUNT(*)
+			FROM product_vendor pv
+			JOIN price pr ON pr.product_vendor_id = pv.id
+			WHERE pr.vendor_id = $1
+		`
```
Comment (edutjie) [inline internal/product/accessor_test.go:465]: I just noticed this query is wrong... What if the filter is on, the countQuery would still be the unfiltered count 🤦‍♂️

File: internal/product/accessor.go [hunk 2]
```diff
@@ -154,32 +143,45 @@ func (p *postgresProductAccessor) GetProductsByVendor(
...
 	}
 	if err := rows.Err(); err != nil {
 		return nil, err
 	}
 
-	countQuery := `SELECT COUNT(*) FROM product_vendor WHERE vendor_id=$1`
+	countQuery := `SELECT COUNT(*)
+		FROM product_vendor pv
+		JOIN price pr ON pr.product_vendor_id = pv.id
+		WHERE pr.vendor_id = $1`
```
Comment (edutjie) [inline internal/product/accessor.go:162]: I just noticed this query is wrong... What if the filter is on, the countQuery would still be the unfiltered count 🤦‍♂️

Comment (valeelim) [reply internal/product/accessor.go:162]: hwemm nice catch edu, good point

**Comment by ferryhieronimus** [comment]
Could you please elaborate on why we should do this revamp?

**Comment by valenciusap17** [comment]
> Could you please elaborate on why we should do this revamp?

I am guessing that it is because of the changes on the table product columns given by our client, CMIIW. Anyway your code looks great Valerian 👍

**Review by valenciusap17 — APPROVED** [review]

**Review by edutjie — APPROVED** [review]
LGTM

**Pull request merged**